cmd/agent/subcommands/analyzelogs/command.go

```diff
@@ -108,27 +133,56 @@ func runAnalyzeLogs(cliParams *CliParams, config config.Component) error {
 }
 
 // Used to make testing easier
-func runAnalyzeLogsHelper(cliParams *CliParams, config config.Component) (chan *message.Message, *launchers.Launchers, pipeline.Provider) {
+func runAnalyzeLogsHelper(cliParams *CliParams, config config.Component, ac autodiscovery.Component) (chan *message.Message, *launchers.Launchers, pipeline.Provider) {
 	configSource := sources.NewConfigSources()
-	wd, err := os.Getwd()
+	waitTime := time.Duration(1) * time.Second
+	waitCtx, cancelTimeout := context.WithTimeout(
+		context.Background(), waitTime)
+	common.LoadComponents(nil, nil, ac, pkgconfigsetup.Datadog().GetString("confd_path"))
+	ac.LoadAndRun(context.Background())
+	allConfigs, err := common.WaitForConfigsFromAD(waitCtx, []string{cliParams.LogConfigPath}, 1, "", ac)
+	cancelTimeout()
 	if err != nil {
-		fmt.Println("Cannot get working directory")
 		return nil, nil, nil
 	}
-	absolutePath := wd + "/" + cliParams.LogConfigPath
-	data, err := os.ReadFile(absolutePath)
-	if err != nil {
-		fmt.Println("Cannot read file path of logs config")
-		return nil, nil, nil
+	var sources []*sources.LogSource
+	sources = nil
+	for _, config := range allConfigs {
+		if config.Name != cliParams.LogConfigPath {
+			continue
+		}
+		sources, err = ad.CreateSources(integration.Config{
+			Provider:   names.File,
+			LogsConfig: config.LogsConfig,
+		})
+		if err != nil {
+			fmt.Println("Cannot create source")
+			return nil, nil, nil
+		}
+		break
 	}
-	sources, err := ad.CreateSources(integration.Config{
-		Provider:   names.File,
-		LogsConfig: data,
-	})
 
-	if err != nil {
-		fmt.Println("Cannot create source")
-		return nil, nil, nil
+	if sources == nil {
```

Comment:
Maybe we should flip this logic around:

if `LogConfigPath` points to a valid path on disk, use that config and don't bother setting up autodiscovery, else if the path is invlaid assume it's a check name and spin up the AD logic. 

So something like (considering my above comment about abstracting out config resolution)

```go

fileConfig, err := resolveFileConfig(cliParams.LogConfigPath)

if err != nil {
    // assume it's a check name and not a file
    configs, err := resolveConfigs(cliParams.LogConfigPath)
    
    if err != nil {
    // print error with some context ex: invlaid path or check name
    return 
    } 
}

// Run analyze logs with configs resolved from one of the previous steps
```
(Pseudocode, feel free to restructure as you see fit)